tutorials/python/lbs/first/first_example.ipynb

```diff
@@ -0,0 +1,438 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# A First Example\n",
+    "\n",
+    "This is a complete simulation transport example. Each aspect of the simulation process is kept to a minimum:\n",
+    "- We use an orthogonal 2D grid;\n",
+    "- We introduce the concept of domain decomposition (\"partitioning\");\n",
+    "- The domain is homogeneous (single material, uniform isotropic external source), vacuum boundary conditions apply;\n",
+    "- The cross sections are given in a text file (with our OpenSn format); we use only one energy group in this example;\n",
+    "- The angular quadrature (discretization in angle) is introduced;\n",
+    "- The Linear Boltzmann Solver (LBS) options are keep to a minimum.\n",
+    "\n",
+    "\n",
+    "Because transport simulations are computationally expensive due to the high dimensional of the phase-space\n",
+    "(physical space, energy, direction), they are often performed using several parallel processes (e.g., CPU cores).\n",
+    "In what follows, we enforce that this first example be run with 4 MPI processes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "plaintext"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import sys"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Check the number of processes\n",
+    "This portion of the lua input is not mandatory. The user is free to delete it and run the code with a different number\n",
+    "of processes. However, to reproduce the graphics below, one needs to run with 4 MPI ranks.\n",
+    "\n",
+    "The lua input performs this following checks:\n",
+    "+ `check_num_procs==nil` will be true when running serially without MPI\n",
+    "+ `number_of_processes ~= num_procs` will be true when the number of MPI processes requested is not equal to the preset value of `4`.\n",
+    "\n",
+    "To run the code, simply type: `mpiexec -n 4 path/to/opensn -i input_filename.lua`\n",
+    "\n",
+    "For more runtime options, type `path/to/opensn -h` for help."
```

Comment:
maybe we should remove this for a first example (no requirements on running it serial or in parallel with an imposed number of ranks=4). i think this is a distraction. (I did it a long time ago but it should not be that way in my opinion)
another comment: there were still mentions of lua.